src/libraries/System.Text.RegularExpressions/tests/RegexParserTests.netcoreapp.cs

```diff
@@ -0,0 +1,148 @@
+        [InlineData("(?<- 0>)", RegexOptions.IgnorePatternWhitespace, RegexParseError.CaptureGroupNameInvalid, 4)]
+        [InlineData("(?<a-0')", RegexOptions.IgnorePatternWhitespace, RegexParseError.CaptureGroupNameInvalid, 6)]
+        [InlineData("(?'a-0>)", RegexOptions.IgnorePatternWhitespace, RegexParseError.CaptureGroupNameInvalid, 6)]
+        [InlineData("(?'-0')", RegexOptions.IgnorePatternWhitespace, null)]
+        [InlineData("(?'a-0')", RegexOptions.IgnorePatternWhitespace, null)]
+        [InlineData("(?<-0", RegexOptions.IgnorePatternWhitespace, RegexParseError.InvalidGroupingConstruct, 5)]
+        [InlineData("(?<-0)", RegexOptions.IgnorePatternWhitespace, RegexParseError.CaptureGroupNameInvalid, 5)]
+        [InlineData("(?<-0>", RegexOptions.IgnorePatternWhitespace, RegexParseError.InsufficientClosingParentheses, 6)]
+        [InlineData(@"(?<cat>cat)\w+(?<dog-()*!@>dog)", RegexOptions.None, RegexParseError.CaptureGroupNameInvalid, 21)]
+        [InlineData(@"(?<cat>cat)\w+(?<dog-catdog>dog)", RegexOptions.None, RegexParseError.UndefinedNamedReference, 27)]
+        [InlineData(@"(?<cat>cat)\w+(?<dog-1uosn>dog)", RegexOptions.None, RegexParseError.CaptureGroupNameInvalid, 22)]
+        [InlineData(@"(?<cat>cat)\w+(?<dog-16>dog)", RegexOptions.None, RegexParseError.UndefinedNumberedReference, 23)]
+        [InlineData(@"cat(?<->dog)", RegexOptions.None, RegexParseError.CaptureGroupNameInvalid, 7)]
+        [InlineData("a{2147483648}", RegexOptions.None, RegexParseError.QuantifierOrCaptureGroupOutOfRange, 12)]
+        [InlineData("a{2147483648,}", RegexOptions.None, RegexParseError.QuantifierOrCaptureGroupOutOfRange, 12)]
+        [InlineData("a{0,2147483647}", RegexOptions.None, null)]
+        [InlineData("a{0,2147483648}", RegexOptions.None, RegexParseError.QuantifierOrCaptureGroupOutOfRange, 14)]
+        // Surrogate pair which is parsed as [char,char-char,char] as we operate on UTF-16 code units.
+        [InlineData("[\uD82F\uDCA0-\uD82F\uDCA3]", RegexOptions.IgnoreCase, RegexParseError.ReversedCharacterRange, 5)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void Parse_NotNetFramework(string pattern, RegexOptions options, object error, int offset = -1)
+        {
+            Parse(pattern, options, error, offset);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsBinaryFormatterSupported))]
+        public void RegexParseException_Serializes()
+        {
+#pragma warning disable RE0001 // Regex issue: Not enough )'s
+            ArgumentException e = Assert.ThrowsAny<ArgumentException>(() => new Regex("(abc|def"));
+#pragma warning restore RE0001 // Regex issue: Not enough )'s
+
+            var bf = new BinaryFormatter();
+            var s = new MemoryStream();
+            bf.Serialize(s, e);
+            s.Position = 0;
+
+            ArgumentException e2 = (ArgumentException)bf.Deserialize(s);
```

Comment:
This should assert that e2's type is ArgumentException.  As written, it could be a type derived from ArgumentException.

Reply:
Good point, though elsewhere in the runtime there's a similar test for this. Could you make the change, please?

(do note that this test was there for ages already, I didn't review it as part of this PR, but good to improve it)

Reply:
@stephentoub: done, a type test was added.